File: dmn-test-cases/signavio/dmn/dmn2java/expected/complex/dmn/example-credit-decision/type/ApplicantImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "applicant"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class ApplicantImpl implements Applicant {
        private String name;
        private java.lang.Number age;
        private java.lang.Number creditScore;
        private List<String> priorIssues;

    public ApplicantImpl() {
    }

    public ApplicantImpl(java.lang.Number age, java.lang.Number creditScore, String name, List<String> priorIssues) {
        this.setAge(age);
        this.setCreditScore(creditScore);
        this.setName(name);
        this.setPriorIssues(priorIssues);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Age")
    public java.lang.Number getAge() {
        return this.age;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Age")
    public void setAge(java.lang.Number age) {
        this.age = age;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Credit score")
    public java.lang.Number getCreditScore() {
        return this.creditScore;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Credit score")
    public void setCreditScore(java.lang.Number creditScore) {
        this.creditScore = creditScore;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Name")
    public String getName() {
        return this.name;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Name")
    public void setName(String name) {
        this.name = name;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Prior issues")
    public List<String> getPriorIssues() {
        return this.priorIssues;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Prior issues")
    public void setPriorIssues(List<String> priorIssues) {
        this.priorIssues = priorIssues;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-signavio/src/test/resources/dmn2java/exported/proto/expected/proto3/kotlin/dmn/example-credit-decision/type/ApplicantImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "applicant"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class ApplicantImpl : Applicant {
    @get:com.fasterxml.jackson.annotation.JsonGetter("Name")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Name")
    override var name: String? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Age")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Age")
    override var age: java.math.BigDecimal? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Credit score")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Credit score")
    override var creditScore: java.math.BigDecimal? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Prior issues")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Prior issues")
    override var priorIssues: List<String?>? = null

    constructor() {
    }

    constructor (age: java.math.BigDecimal?, creditScore: java.math.BigDecimal?, name: String?, priorIssues: List<String?>?) {
        this.age = age
        this.creditScore = creditScore
        this.name = name
        this.priorIssues = priorIssues
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
